bson/bsonoptions/string_codec_options.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsonoptions
+
+// StringCodecOptions represents all possible options for time.Time encoding and decoding.
+type StringCodecOptions struct {
+	ObjectIDAsHex *bool // Specifies if we should decode ObjectID as the hex value. Defaults to true.
+}
+
+// StringCodec creates a new *StringCodecOptions
+func StringCodec() *StringCodecOptions {
+	return &StringCodecOptions{}
+}
+
+// SetObjectIDAsHex specifies if we should decode ObjectID as the hex value. Defaults to true.
```

Comment:
We should document what happens if this is set to false. For example, "SetObjectIDAsHex specifies if object IDs should be decoded as their hex representation. If false, a string made from the raw object ID bytes will be used. Defaults to true."

Also, given that this is part of the mgocompat epic, should we maybe document that mgo defaults this to false?

Reply:
Done. I don't want to add the mgocompat documentation to the main bson stuff, that might be confusing.